controllers/ManagementController.php

```diff
@@ -195,11 +201,24 @@ private function createAccount(CreateAccountForm $form)
         $user->is_validated = false;
         $user->is_active = true;
         $user->role = $form->role;
+        $token_try = 0;
+
+        do {
+            $token = $this->createUserToken();
+            $token_try++;
+        } while (!is_null(User::findOne(['token' => $token])) AND $token_try < 10);
+
+        if($token_try == 10) {
+            throw new CannotCreateTokenException();
+        }
```

Comment:
Even if it's unlikely, there may be a case where you throw this exception when the token creation has succeeded :
- Your loop failed 9 times, so $tokenTry is 9
- You make a 10th iteration, you succeed to create a user token
- You increments $tokenTry (line 207)
- After the loop, $tokenTry is 10, so you throws the exception, whereas the token is well created

Reply:
That's true, I'll try to figure out something more accurate, and that works in all cases